ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -82,27 +88,31 @@ public void copyAndVerify(Path destRoot, List<ReplChangeManager.FileInfo> srcFil
     }
     FileSystem sourceFs = srcFiles.get(0).getSrcFs();
     boolean useRegularCopy = regularCopy(sourceFs, srcFiles);
+    ExecutorService executorService = null;
     try {
       if (useRegularCopy || readSrcAsFilesList) {
+        executorService = getExecutorService();
```

Comment:
Move this to line number 101. It will avoid unnecessary ThreadPool creation when it is not needed.